atomic_reactor/plugins/exit_delete_from_registry.py

```diff
@@ -42,16 +42,70 @@ def __init__(self, tasker, workflow, registries):
 
         self.registries = deepcopy(registries)
 
+    def request_delete(self, url, manifest, insecure, auth):
+        response = requests.delete(url, verify=not insecure, auth=auth)
+
+        if response.status_code == requests.codes.ACCEPTED:
+            self.log.info("deleted manifest %s", manifest)
+            return True
+        elif response.status_code == requests.codes.NOT_FOUND:
+            self.log.warning("cannot delete %s: not found", manifest)
+        elif response.status_code == requests.codes.METHOD_NOT_ALLOWED:
+            self.log.warning("cannot delete %s: image deletion disabled on registry",
+                             manifest)
+        else:
+            msg = "failed to delete %s: %s" % (manifest, response.reason)
+            self.log.error("%s\n%s", msg, response.text)
+            raise PluginFailedException(msg)
+
+        return False
+
+    def make_manifest(self, registry, repo, digest):
+        return "{registry}/{repo}@{digest}".format(**vars())
+
+    def make_url(self, registry, repo, digest):
+        return "{registry}/v2/{repo}/manifests/{digest}".format(**vars())
+
+    def get_worker_digests(self):
+        """
+         If we are being called from an orchestrator build, collect the worker
+         node data and recreate the data locally.
+        """
+        try:
+            builds = self.workflow.build_result.annotations['worker-builds']
+        except(TypeError, KeyError):
+            # This annotation is only set for the orchestrator build.
+            # It's not present, so this is a worker build.
+            return {}
+
+        worker_digests = {}
+
+        for plat, annotation in builds.items():
+            digests = annotation['digests']
+            self.log.debug("build %s has digests: %s" % (plat, digests))
+
+            for digest in digests:
+                reg = digest['registry']
+                if reg in worker_digests:
+                    worker_digests[reg].append(digest)
+                else:
+                    worker_digests[reg] = [digest]
```

Comment:
Hmm, you skipped the part where 'reg = digest['registry']', so the above won't quite work that way. :-)

I think you need something like
worker_digests = {}
worker_digests = [{ digest['registry'] : [digest] } for digest in digests]
or something like that.  But you need to handle duplicate digest['registry'] but different digest.

I am sure there is a pythonic way to do that, I just don't know how. :-(